tools/bin/ci_check_dependency.py

```diff
@@ -0,0 +1,84 @@
+import sys
+import os
+import os.path
+
+CONNECTOR_FOLDER = "airbyte-integrations/connectors/"
+CONNECTOR_PATH = "./airbyte-integrations/connectors/"
+WHITE_LIST = ["/src/main/", "build.gradle"]
+
+
+def main():
+    git_diff_file_path = ' '.join(sys.argv[1:])
+
+    # Get changed connectors files
+    connectors_changed_files = get_connectors_changed_files(git_diff_file_path)
+    # Get changed connectors
+    changed_connectors = get_changed_connectors(connectors_changed_files)
+
+    # Get all existing connectors
+    all_connectors = get_all_connectors()
+
+    # Getting all build.gradle file
+    all_build_gradle_files = get_connectors_gradle_files(all_connectors)
+
+    # Try to find dependency in build.gradle file
+    depended_connectors = list(set(get_depended_connectors(changed_connectors, all_build_gradle_files)))
+
+    write_report(depended_connectors)
+
+
+def get_connectors_changed_files(path):
+    changed_connectors_files = []
+    with open(path) as file:
+        for line in file:
+            if CONNECTOR_FOLDER in line:
```

Comment:
If I understand this code correctly, it iterates through all changed files returned by `git diff` and returns only the files under `airbyte-integrations/connectors/` folder. If that's so, this step can be avoided by using a more specific `git diff` command:
```
git diff --name-only remotes/origin/master -- airbyte-integrations/connectors/ origin/${{ steps.extract_branch.outputs.branch }} -- airbyte-integrations/connectors/ > ./changed_files.txt
```

Reply:
Good point. Since I'll need to analyse `airbyte-integrations/bases` then I think the right git diff will looks like:
`git diff  --name-only  remotes/origin/master...origin/${{ steps.extract_branch.outputs.branch }} -- airbyte-integrations/`